src/utils/ComponentsTypes.ts

```diff
@@ -13,7 +13,11 @@ export class ComponentsTypes {
 
   public static getAllFacetsElements(root: HTMLElement | Dom) {
     const selectors = ComponentsTypes.allFacetsClassname.map(className => `.${className}`).join(', ');
-    return $$(root as HTMLElement).findAll(selectors);
+    const hasNoFacetChild = (element: HTMLElement) => !$$(element).findAll(selectors).length;
+
+    return $$(root as HTMLElement)
+      .findAll(selectors)
+      .filter(hasNoFacetChild);
```

Comment:
This should be easy to test. You can build a root with two facets where:
a) one is a child of the other => expect to get only 1 facet back.
b) they are siblings and children of the root => expect to get both back.

Reply:
It's tested already by checking the length of the facets returned vs the expected length. Without that line it would fail.
I could add another test just for that facet but I believe it would be a bit redundant.

Reply:
Updated the test description so it's more detailed